hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2966,6 +2966,19 @@ public String getRpcPort() {
   }
 
   @Override
+
+  public String getOmRatisRoles() {
+    List<ServiceInfo> serviceList = null;
+    int port = omNodeDetails.getRatisPort();
+    try {
+      serviceList = getServiceList();
+    } catch (IOException e) {
```

Comment:
Using getServiceList() won't work because according to it's [logic](https://github.com/apache/ozone/blob/e6b6a842a19e5c64e62e03b039aa05fbe218ddbf/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3023) , the node that calls this method is the leader and since in the web UI case every OM node will make this call, it will show every node as leader

Reply:
Check this out https://github.com/apache/ozone/pull/3401/files

Reply:
Thanks for the help @sadanand48, I have made changes to the patch do take a look!